From: Colin Watson <cjwatson@debian.org>
Date: Mon, 17 Feb 2025 00:39:07 +0000
Subject: Add CA constraint to test CA

Otherwise recent versions of OpenSSL reject it as an invalid CA
certificate (at least once the test certificates are regenerated).

Fixes: #6896

Forwarded: https://github.com/psf/requests/pull/6897/commits/507409661335bd3dd8a7e39f04d07b42e519becc
Bug: https://github.com/psf/requests/issues/6896
Bug-Debian: https://bugs.debian.org/1091503
Last-Update: 2025-02-17
---
 tests/certs/expired/ca/ca.cnf | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/certs/expired/ca/ca.cnf b/tests/certs/expired/ca/ca.cnf
index 8c4b823..1443cb9 100644
--- a/tests/certs/expired/ca/ca.cnf
+++ b/tests/certs/expired/ca/ca.cnf
@@ -4,9 +4,13 @@ prompt = no
 default_md = sha256
 encrypt_key = no
 distinguished_name = dn
+x509_extensions = v3_ca
 
 [dn]
 C = US                            # country code
 O = Python Software Foundation    # organization
 OU = python-requests              # organization unit/department
 CN = Self-Signed Root CA          # common name / your cert name
+
+[v3_ca]
+basicConstraints = critical, CA:true
